core/src/main/java/io/questdb/griffin/engine/functions/groupby/FirstStrGroupByFunction.java

```diff
@@ -100,12 +140,18 @@ public boolean isScalar() {
     @Override
     public void pushValueTypes(ArrayColumnTypes columnTypes) {
         this.valueIndex = columnTypes.getColumnCount();
-        columnTypes.add(ColumnType.BOOLEAN);
+        columnTypes.add(ColumnType.INT); // sink index
+        columnTypes.add(ColumnType.BOOLEAN); // null flag
```

Comment:
nit: a bit of overhead to store not null flag if we only compute first record. I would understand the need in first not null but here it looks redundant, simply storing null `DirectCharSink` would be good enough.